.github/workflows/gap.yml

```diff
@@ -0,0 +1,75 @@
+name: CI with GAP
+
+on:
+  push:
+    branches:
+      - 'master'
+      - 'release-*'
+    tags: '*'
+  pull_request:
+
+concurrency:
+  # group by workflow and ref; the last slightly strange component ensures that for pull
+  # requests, we limit to 1 concurrent job, but for the default repository branch we don't
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.ref_name != github.event.repository.default_branch || github.run_number }}
+  # Cancel intermediate builds, but only if it is a pull request build.
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+
+jobs:
+  test:
+    name: Julia ${{ matrix.julia-version }} - GAP ${{ matrix.gap-version }} - ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 20
+    continue-on-error: ${{ matrix.julia-version == 'nightly' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        gap-version:
+          - 'master'
+          - 'stable-4.13'
+          - 'v4.13.0'       # latest release
```

Comment:
I am not sure that's all too helpful. Point is: we may evolve some internals of the GC integration requiring changes both in GAP and in GAP.jl, and it may be impractical to also keep supporting the last stable release. I'd just drop this, to reduce the overall number of jobs.

Or at the very least, drop the macOS tests against that, we really only need to test macOS against Julia 1.10 (resp latest 1.x) with GAP master and stable-4.13

Reply:
Thinking again about it: the latest release is already tested via the latest GAP_jll release (more or less at least), so the version bump in GAP_jll and GAP_lib_jll can be used to concurrently change GAP and GAP.jll as you described.